cmd/commands/service/command.go

```diff
@@ -51,51 +52,83 @@ var (
 )
 
 // NewCommand function creates service command
-func NewCommand(licenseCommandName string) *cli.Command {
+func NewCommand(licenseCommandName string, defaultServiceTypes []string) *cli.Command {
 	var di cmd.Dependencies
-
 	command := &cli.Command{
-		Name:      serviceCommandName,
-		Usage:     "Starts and publishes service on Mysterium Network",
-		ArgsUsage: " ",
+		Name:        serviceCommandName,
+		Usage:       "Starts and publishes services on Mysterium Network",
+		ArgsUsage:   " ",
+		Subcommands: getSubcommands(&di, licenseCommandName, defaultServiceTypes),
 		Action: func(ctx *cli.Context) error {
-			if !ctx.Bool(agreedTermsConditionsFlag.Name) {
-				printTermWarning(licenseCommandName)
-				os.Exit(2)
-			}
-
-			errorChannel := make(chan error, 3)
-
-			if err := di.Bootstrap(cmd.ParseFlagsNode(ctx)); err != nil {
-				return err
-			}
-			go func() { errorChannel <- di.Node.Wait() }()
-
-			if err := di.ServiceManager.Start(parseFlags(ctx)); err != nil {
-				return err
-			}
-			go func() { errorChannel <- di.ServiceManager.Wait() }()
-
-			cmd.RegisterSignalCallback(func() { errorChannel <- nil })
-
-			err := <-errorChannel
-			switch err {
-			case service.ErrorLocation:
-				printLocationWarning("myst")
-				return nil
-			default:
-				return err
-			}
+			return runServices(ctx, &di, licenseCommandName, defaultServiceTypes)
 		},
 		After: func(ctx *cli.Context) error {
 			return di.Shutdown()
 		},
 	}
+
 	registerFlags(&command.Flags)
 
 	return command
 }
 
+func runServices(ctx *cli.Context, di *cmd.Dependencies, licenseCommandName string, serviceTypes []string) error {
+	if !ctx.Bool(agreedTermsConditionsFlag.Name) {
+		printTermWarning(licenseCommandName)
+		os.Exit(2)
+	}
+
+	errorChannel := make(chan error, 2+len(serviceTypes))
+
+	if err := di.Bootstrap(cmd.ParseFlagsNode(ctx, serviceTypes)); err != nil {
+		return err
+	}
+	go func() { errorChannel <- di.Node.Wait() }()
+
+	for _, serviceType := range serviceTypes {
+		options, err := parseFlagsByServiceType(ctx, serviceType)
+		if err != nil {
+			return err
+		}
+		err = di.ServiceRunner.StartServiceByType(serviceType, options, errorChannel)
```

Comment:
the error here tells us if the startup of services was successful. The channel is then used to report runtime errors that occur in the underlying service.

Reply:
Hmm. Do we really need to report those errors upstairs? Startup errors are pretty much ok, but runtime errors - well I am not sure, what happens in case of that error - we shutdown whole node? What if other services run perfectly? Maybe some kind of retry mechanism is needed? Don't forget that service will be managed through api in the future - not only by args on startup.